### PR TITLE
fix: EB-(761) Live edit button doesn't go away on page navigation

### PR DESCRIPTION
**Pull request by contentstackMridul**
fix: EB-(761) Live edit button doesn't go away on page navigation

### CODE REVIEW

File: src/live-preview.ts [hunk 1]
```diff
@@ -178,6 +197,43 @@ export default class LivePreview {
         }
     }
 
+    private removeEditButtonStyle(e: MouseEvent) {
+        if (!this.currentElementBesideTooltip || !this.tooltip) return false;
```
Comment (Deepak-Kharah) [inline src/live-preview.ts:201]: The size of this file is increasing. Let's consider moving these functions into new files.

File: src/live-preview.ts [hunk 2]
```diff
@@ -382,9 +438,20 @@ export default class LivePreview {
             this.config.editButton.enable
         ) {
             const tooltip = document.createElement("button");
+            const tooltipInnerContainer = document.createElement("div");
+            tooltipInnerContainer.classList.add("cslp-tooltip-inner-container");
+            tooltip.classList.add("cslp-tooltip");
```
Comment (Deepak-Kharah) [inline src/live-preview.ts:443]: Why do we need this element?

File: src/live-preview.ts [hunk 3]
```diff
@@ -394,13 +461,18 @@ export default class LivePreview {
...
+            tooltip.appendChild(tooltipInnerContainer);
             this.tooltip = tooltip;
         }
         this.updateTooltipPosition();
     };
 
+    private toggleHoveringOnEditButton = (isHoveringOnTooltip: boolean) => {
+        this.isHoveringOnTooltip = isHoveringOnTooltip;
+    };
+
```
Comment (Deepak-Kharah) [inline src/live-preview.ts:475]: We can consider moving it to another file.

**Review by Deepak-Kharah — CHANGES_REQUESTED** [review]
We should refine the code more

File: src/live-preview.ts (hunk 2 above)
Comment (contentstackMridul) [reply src/live-preview.ts:443]: This extra element ensures that the edit button does not hide until the customer has completely moved away from the currently hovered element and its associated edit button, without impacting the overall look of the button.

File: src/live-preview.ts (hunk 3 above)
Comment (contentstackMridul) [reply src/live-preview.ts:475]: sure, will move this function to utils files

File: src/live-preview.ts (hunk 1 above)
Comment (contentstackMridul) [reply src/live-preview.ts:201]: sure, will move this function to utils files

**Review by Deepak-Kharah — APPROVED** [review]
The changes look good

**Pull request merged**